src/libraries/System.IO.Pipes/tests/UnifiedErrorHandlingTests.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Pipes.Tests
+{
+    public static class UnifiedErrorHandlingTests
+    {
+        [Fact]
+        public static void WhenAnonymousPipeServerIsClosedAnonymousPipeClientReadReturnsZero()
```

Comment:
@stephentoub can we meet somewhere in the middle?

```diff
- WhenAnonymousPipeServerIsClosedAnonymousPipeClientReadReturnsZero
+ When_AnonymousPipeServer_IsClosed_AnonymousPipeClient_ReadReturnsZero
````

Since we are using the `[Fact]` to annotate the tests I am trying to name in the way that they describe the fact they are testing.